From 9c71e9cb778958f114a236ad46057876aad10894 Mon Sep 17 00:00:00 2001
From: Gabor Juhos <juhosg@openwrt.org>
Date: Thu, 1 Aug 2013 14:42:05 +0200
Subject: [PATCH 17/40] rt2x00: rt2800lib: enable RF3853 support

Signed-off-by: Gabor Juhos <juhosg@openwrt.org>
---
 drivers/net/wireless/rt2x00/rt2800lib.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/wireless/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/rt2x00/rt2800lib.c
@@ -7329,6 +7329,7 @@ static int rt2800_init_eeprom(struct rt2
 	case RF3290:
 	case RF3320:
 	case RF3322:
+	case RF3853:
 	case RF5360:
 	case RF5370:
 	case RF5372:
