From 5bd1e6ff9f7b245f609e08618cfd3b74a4131799 Mon Sep 17 00:00:00 2001
From: Gabor Juhos <juhosg@openwrt.org>
Date: Sun, 24 Mar 2013 19:26:26 +0100
Subject: [PATCH 23/51] rt2x00: rt2800lib: enable support for RT3883

Signed-off-by: Gabor Juhos <juhosg@openwrt.org>
---
 drivers/net/wireless/rt2x00/rt2800lib.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/wireless/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/rt2x00/rt2800lib.c
@@ -7907,6 +7907,7 @@ static int rt2800_probe_rt(struct rt2x00
 	case RT3390:
 	case RT3572:
 	case RT3593:
+	case RT3883:
 	case RT5390:
 	case RT5392:
 	case RT5592:
